cli.go

```diff
@@ -60,39 +67,52 @@ func main() {
 	}
 }
 
-func PromptDiscord(action string) *DiscordInstall {
-	fmt.Println("Please choose a Discord install to", action)
-	for i, discord := range discords {
-		install := discord.(*DiscordInstall)
-		fmt.Printf("[%d] %s%s (%s)\n", i+1, Ternary(install.isPatched, "(PATCHED) ", ""), install.path, install.branch)
-	}
-	fmt.Printf("[%d] Custom Location\n", len(discords)+1)
-
-	var choice int
-	for {
-		fmt.Printf("> ")
-		if _, err := fmt.Scan(&choice); err != nil {
-			fmt.Println("That wasn't a valid choice")
-			continue
-		}
-
-		choice--
-		if choice >= 0 && choice < len(discords) {
-			return discords[choice].(*DiscordInstall)
-		}
-
-		if choice == len(discords) {
-			var custom string
-			fmt.Print("Custom Discord Install: ")
-			if _, err := fmt.Scan(&custom); err == nil {
-				if discord := ParseDiscord(custom, ""); discord != nil {
-					return discord
-				}
-			}
-		}
-
-		fmt.Println("That wasn't a valid choice")
-	}
+func PromptDiscord(action string, client *DiscordInstall, dir string) *DiscordInstall {
+    if client != nil {
+        return client
+    }
+    fmt.Println("Please choose a Discord install to", action)
+    for i, discord := range discords {
+        install := discord.(*DiscordInstall)
+        fmt.Printf("[%d] %s%s (%s)\n", i+1, Ternary(install.isPatched, "(PATCHED) ", ""), install.path, install.branch)
+    }
+    if dir != "" {
+        fmt.Printf("[%d] %s\n", len(discords)+1, dir)
+    } else {
+        fmt.Printf("[%d] Custom Location\n", len(discords)+1)
+    }
+
+    var choice int
+    for {
+        fmt.Printf("> ")
+        if _, err := fmt.Scan(&choice); err != nil {
+            fmt.Println("That wasn't a valid choice")
+            continue
+        }
+
+        choice--
+        if choice >= 0 && choice < len(discords) {
+            return discords[choice].(*DiscordInstall)
+        }
+
+        if choice == len(discords) {
+            if dir != "" {
+                if discord := ParseDiscord(dir, ""); discord != nil {
+                    return discord
+                }
+            } else {
+                var custom string
+                fmt.Print("Custom Discord Install: ")
+                if _, err := fmt.Scan(&custom); err == nil {
+                    if discord := ParseDiscord(custom, ""); discord != nil {
+                        return discord
+                    }
+                }
+            }
+        }
```

Comment:
**comment (non-blocking):** It looks like you changed the indentation formatting in this block, which the maintainer may have opinions about.